### PR TITLE
Postcode optional again

### PR DESCRIPTION
**Pull request by afiore**
- Revert mandatory postcode
- Send a bogus value to Identity when user-supplied postcode is blank

### CODE REVIEW

**Comment by afiore** [comment]
@ostapneko

File: app/services/IdentityService.scala [hunk 1]
```diff
@@ -183,9 +183,18 @@ object IdentityApiClient extends IdentityApiClient with LazyLogging {
...
     val endpoint = authoriseCall(WS.url(s"$identityEndpoint/guest"))
 
-    guestData => endpoint.post(guestData: JsObject)
-      .withWSFailureLogging(endpoint)
-      .withCloudwatchMonitoringOfPost
+    guestData => {
+      // When postcode is blank (e.g. Ireland), force-feed identity with a bogus postcode
+      val idSafeGuestData =
+        if (guestData.address.postCode.isEmpty)
+          guestData.copy(address = guestData.address.copy(postCode = "---"))
```
Comment (ostapneko) [inline app/services/IdentityService.scala]: We also need to change the way this information is read from identity

**Comment by ostapneko** [comment]
:+1:

File: app/services/IdentityService.scala (hunk 1 above)
Comment (afiore) [inline app/services/IdentityService.scala]: The user never returns to the checkout page, so I haven't done that

**Pull request merged**